### ISSUE TITLE
Usage Questions and Feedback

### ISSUE THREAD

**Issue by robertcaretta** [microsoft/MSCloudLoginAssistant#124]
1) It says it the module code notes that "MSCloudLoginAssistant.psm1" is for connecting to SharePoint Online and SharePoint PnP. However not seeing anything SPO specific, just PnP as platform/workload. Can you clarify on this?

2) Why are there two cmdlets for connecting? Is the proper usage of this to **always** just use Test-MSCloudLogin?

3) What is the reasoning being using Test-MSCloudLogin to pass parameters off to Connect-M365Tenant?   

4) Could you provide more detailed examples? I haven't been able to establish a connection with the PnP, even though the service account I am using has proper access rights. I've tried using the TenantId and ConnectionUrl for the SP admin center. I have the PnP powershell module installed in the PS 7.2 location.

![image](https://user-images.githubusercontent.com/1613085/187555752-3d40aa5c-fac0-4f9b-826e-4a02bb5d1bb6.png)

![image](https://user-images.githubusercontent.com/1613085/187555630-a158cc05-fafb-487e-80ce-c269f70589ef.png)

5) Would it be possible to add a debug mode option which outputs more information to the terminal? Verbose doesn't seem to do much.

**Comment by andikrueger** [comment]
Just a few quick answers on the questions:

1. This module used to use the SharePoint Online Management Module (see: https://github.com/microsoft/MSCloudLoginAssistant/blob/4dfb6e6947212706a1b0b143beb32cde4bea79f9/Modules/MSCloudLoginAssistant/Workloads/SharePointOnline.psm1). This module was removed in favor of PnP. The description is not entirely correct nor incorrect.
2. This looks also like a relict to the early usage within the Microsoft365DSC module. The `Test-MSCloudLogin` function was used in previous versions of M365DSC. Nowadays, M365DSC calls Connect-M365Tenant.
3. See 2..


With Ps 7.2 there are sometimes issues with the location of where modules are stored. Did you install MSCloudloginAssistant in PS 7.2 or in a previous version? Same applies for PnP.

For what kind of additional information are you looking for?

**Comment by robertcaretta** [comment]
1. I would say it's not accurate. There are two separate modules SPO and PnP. They are similar but there are differences  in functionality and maintainers.

2. So is the intent of this module to be strictly used as a dependency for [M365DSC](https://microsoft365dsc.com/)? 

I installed MSCloudLoginAssistant to 5.1 and had PnP installed. Then I uninstalled PnP and reinstalled it to 5.1.  On line 35 of PnP.psm1 it is using this command. _Import-Module PnP.PowerShell -UseWindowsPowerShell -Global -Force | Out-Null_   So I would presume by that PnP needs to be installed in 5.1.  Once I had everything installed into 5.1 it seemed to work, even running in a 7.2 window.

I guess I'm looking for general guidance here on how this module should be used, if at all. Is this a module that is intended to be used directly? Or is it intended to be a dependency of M365DSC? The documentation is not clear on intent or usage examples.
